### PR TITLE
docs: update php type and replace annotations with PHP attributes

### PR DESCRIPTION
**Pull request by StephaneBullier**

### CODE REVIEW

**Comment by alanpoulain** [comment]
I don't understand why you have done another PR, should we consider this PR the main one and closing the other one?

File: core/subresources.md [hunk 1]
```diff
@@ -59,28 +49,18 @@ use ApiPlatform\Core\Annotation\ApiResource;
...
+    private ?int $id = null;
+
+    #[ORM\Column(type: 'text')]
     public $content;
 
-    /**
-     * @ORM\OneToOne(targetEntity="Answer", inversedBy="question")
-     * @ORM\JoinColumn(referencedColumnName="id", unique=true)
-     */
+    #[ORM\OneToOne(targetEntity: Answer::class, insersedBy: 'question')]
```
Comment (alanpoulain) [inline core/subresources.md:62]:
```suggestion
    #[ORM\OneToOne(targetEntity: Answer::class, inversedBy: 'question')]
```

File: core/subresources.md [hunk 2]
```diff
@@ -59,28 +49,18 @@ use ApiPlatform\Core\Annotation\ApiResource;
 use ApiPlatform\Core\Annotation\ApiSubresource;
 use Doctrine\ORM\Mapping as ORM;
 
-/**
- * @ORM\Entity
- */
+#[ORM\Entity()]
```
Comment (alanpoulain) [inline core/subresources.md:52]:
```suggestion
#[ORM\Entity]
```

File: admin/handling-relations.md [hunk 3]
```diff
@@ -156,23 +156,15 @@ namespace App\Entity;
...
-     * @ORM\Id
-     */
-    public $id;
-
-    /**
-     * @ORM\ManyToOne(targetEntity=Book::class, inversedBy="reviews")
-     */
-    public $book;
+    #[ORM\Id, ORM\Column, ORM\GeneratedValue]
+    public ?int $id = null;
```
Comment (alanpoulain) [inline admin/handling-relations.md:164]: Why nullable?

File: core/getting-started.md [hunk 4]
```diff
@@ -117,32 +111,22 @@ use Symfony\Component\Validator\Constraints as Assert;
...
-    /**
-     * @ORM\Column(type="text")
-     */
+    #[ORM\Column(type: 'text')]
     public string $description = '';
 
-    /**
-     * @ORM\Column(type="float")
-     */
+    #[ORM\Column(type: 'float')]
```
Comment (alanpoulain) [inline core/getting-started.md:125]: Is this necessary?

File: core/openapi.md [hunk 5]
```diff
@@ -150,16 +144,14 @@ class Product // The class name will be used to name exposed resources
...
-    /**
-     * @ORM\Column
-     */
+    #[ORM\Column] 
     #[Assert\DateTime]
     #[ApiProperty(
         openapi_context: ["type" => "string", "format" => "date-time"]
     )]
-    public $timestamp;
+    public \DateTimeInterface $timestamp;
```
Comment (alanpoulain) [inline core/openapi.md:154]: To test: I don"t think the validation will be performed if there is a PHP type.

File: core/subresources.md [hunk 6]
```diff
@@ -20,27 +20,17 @@ namespace App\Entity;
...
+    #[ORM\Id, ORM\Column, ORM\GeneratedValue]
+    private ?int $id = null;
+
+    #[ORM\Column(type: 'text')]
+    public string $content;
 
-    /**
-     * @ORM\OneToOne(targetEntity="Question", mappedBy="answer")
-     */
+    #[ORM\OneToOne(targetEntity: Question::class, mappedBy: 'answer')]
```
Comment (alanpoulain) [inline core/subresources.md:33]: Why not having the PHP type here?

File: core/subresources.md [hunk 7]
```diff
@@ -59,28 +49,18 @@ use ApiPlatform\Core\Annotation\ApiResource;
...
+    private ?int $id = null;
+
+    #[ORM\Column(type: 'text')]
     public $content;
 
-    /**
-     * @ORM\OneToOne(targetEntity="Answer", inversedBy="question")
-     * @ORM\JoinColumn(referencedColumnName="id", unique=true)
-     */
+    #[ORM\OneToOne(targetEntity: Answer::class, inversedBy: 'question')]
```
Comment (alanpoulain) [inline core/subresources.md:62]: PHP type?

File: core/validation.md [hunk 8]
```diff
@@ -26,34 +26,30 @@ use Symfony\Component\Validator\Constraints as Assert; // Symfony's built-in con
...
 #[ApiResource]
 class Product
 {
     /**
-     * @var int The id of this product.
-     *
-     * @ORM\Id
-     * @ORM\GeneratedValue
-     * @ORM\Column(type="integer")
+     * @var The id of this product.
```
Comment (alanpoulain) [inline core/validation.md:35]: I don't think the PHPDoc will work like this.

File: core/validation.md [hunk 9]
```diff
@@ -26,34 +26,30 @@ use Symfony\Component\Validator\Constraints as Assert; // Symfony's built-in con
...
-    private $id;
+    #[ORM\Id, ORM\Column, ORM\GeneratedValue]
+    private ?int $id = null;
 
     /**
-     * @var string The name of the product
-     *
-     * @Assert\NotBlank
-     * @ORM\Column
+     * @var The name of the product
```
Comment (alanpoulain) [inline core/validation.md:41]: I don't think the PHPDoc will work like this.

File: distribution/index.md [hunk 10]
```diff
@@ -373,7 +373,7 @@ Modify these files as described in these patches:
 +/**
 + * A book.
 + *
-+ * @ORM\Entity
++ * #[ORM\Entity]
```
Comment (alanpoulain) [inline distribution/index.md:376]: This file is weird.

File: core/validation.md (hunk 8 above)
Comment (StephaneBullier) [reply core/validation.md:35]: I don't understand exactly the problem.
In Doctrine docs (https://www.doctrine-project.org/2021/05/24/orm2.9.html) you can read this : #[ORM\Id, ORM\Column, ORM\GeneratedValue]

**Comment by soyuka** [comment]
Thanks @StephaneBullier

**Pull request merged**